src/main/java/com/cdancy/jenkins/rest/parsers/OptionalFolderPathParser.java

```diff
@@ -20,9 +20,15 @@ public String apply(Object optionalFolderPath) {
 
         String folderPath = String.class.cast(optionalFolderPath);
         String amendedPath = "";
-        if(folderPath.startsWith("/") || folderPath.endsWith("/")) {
-            throw new RuntimeException("Incorrect folder Path format - " + folderPath);
+
+        if(folderPath.startsWith("/")) {
+            folderPath = folderPath.replaceAll("^/+","");
+        }
+
+        if(folderPath.endsWith("/")) {
+            folderPath = folderPath.replaceAll("/+$","");
         }
+
```

Comment:
@Dhivyaa21 looks good. With all the string manipulation would it be better to use a StringBuilder here? It would remove the need to use regex's via the `replaceAll` method and instead we could just remove the first and last character if present. Would also speed up the concatenation of strings below.